program/src/state.rs

```diff
@@ -811,7 +811,7 @@ impl HealthCache {
                     &mango_cache.root_bank_cache[i],
                     mango_cache.price_cache[i].price,
                     i,
-                    &if *open_orders_ais[i].key == Pubkey::default() {
+                    &if mango_account.spot_open_orders[i] == Pubkey::default() {
                         None
                     } else {
                         Some(load_open_orders(&open_orders_ais[i])?)
```

Comment:
This would work if an account needs no open orders at all. But imagine they need the oo account for i=4 and pass only that. It looks like this would try to look up `open_orders_ais[4]`.

Reply:
yes, it only supports truncating the end right now, not leaving out spots in the middle